tests/hardhat/Fork/comptrollerXVSAddressesUpgrade.ts

```diff
@@ -0,0 +1,132 @@
+import { smock } from "@defi-wonderland/smock";
+import { impersonateAccount } from "@nomicfoundation/hardhat-network-helpers";
+import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers";
+import chai from "chai";
+import { ethers, network } from "hardhat";
+
+import { BEP20, Diamond, RewardFacet, SetterFacet } from "../../../typechain";
+
+const { expect } = chai;
+chai.use(smock.matchers);
+
+const Owner = "0x939bd8d64c0a9583a7dcea9933f7b21697ab6396";
+const OLD_SETTER_FACET = "0xF2b7D75557B75a878E997934014E95Dd089B5f24";
+const OLD_REWARD_FACET = "0x71e7AAcb01C5764A56DB92aa31aA473e839d964F";
+const UNITROLLER = "0xfD36E2c2a6789Db23113685031d7F16329158384";
+const zeroAddr = "0x0000000000000000000000000000000000000000";
+
+export async function setForkBlock(blockNumber: number) {
+  await network.provider.request({
+    method: "hardhat_reset",
+    params: [
+      {
+        forking: {
+          jsonRpcUrl: process.env[`ARCHIVE_NODE_${process.env.FORKED_NETWORK}`],
+          blockNumber: blockNumber,
+        },
+      },
+    ],
+  });
+}
+
+const forking = (blockNumber: number, fn: () => void) => {
+  describe(`Reward and Setter facet upgrade check At block #${blockNumber}`, () => {
+    before(async () => {
+      await setForkBlock(blockNumber);
+    });
+    fn();
+  });
+};
+
```

Comment:
We can import these functions from [utils.ts](https://github.com/VenusProtocol/venus-protocol/blob/7d3e10178e51982a7474a2831fc38e8f89a0faeb/tests/hardhat/Fork/utils.ts).